### PR TITLE
wallet, simplewallet: Drop support for mine-to-use RPC system

### PR DESCRIPTION
**Pull request by jeffro256**
The main thing to test here is new wallet UX with old RPC mine-to-use node. Using post-PR wallet with pre-PR node should generate error message specific to this PR and not generic "command failed" mesages.

### CODE REVIEW

**Comment by jeffro256** [comment]
https://github.com/monero-project/monero/issues/8722

**Comment by plowsof** [comment]
New wallet connecting old rpc mine-to-use node error msg: (i guess the refresh thread line could be dropped, and im not sure about the See . Block received: 0 being on the same line)
```
Starting refresh...
Error: refresh failed: Daemon requires deprecated RPC payment. See . Blocks received: 0
Background refresh thread started
[wallet 437Jhp (out of sync)]: 
```
:+1:

**Comment by jeffro256** [comment]
@plowsof Thanks for testing. That error message was a typo, I meant to put the link to the relevant GH issue.

File: src/wallet/wallet_rpc_server.cpp [hunk 1]
```diff
@@ -4517,7 +4517,6 @@ class t_daemon
 
       const auto arg_wallet_file = wallet_args::arg_wallet_file();
       const auto arg_from_json = wallet_args::arg_generate_from_json();
-      const auto arg_rpc_client_secret_key = wallet_args::arg_rpc_client_secret_key();
```
Comment (rbrunner7) [inline src/wallet/wallet_rpc_server.cpp:4520]: I am a bit confused: If I checked correctly you let a lot of RPC payment related stuff stand in the wallet RPC server, but out all possible things to delete you only delete things related to this key.

If that's really the case, what's the idea here?

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -540,10 +529,9 @@ void simple_wallet::handle_transfer_exception(const std::exception_ptr &e, bool
     {
       std::rethrow_exception(e);
     }
-    catch (const tools::error::payment_required&)
+    catch (const tools::error::deprecated_rpc_access&)
     {
-      fail_msg_writer() << tr("Payment required, see the 'rpc_payment_info' command");
-      m_need_payment = true;
+      fail_msg_writer() << tr("Daemon requires deprecated RPC payment. See https://github.com/monero-project/monero/issues/8722");
```
Comment (rbrunner7) [inline src/simplewallet/simplewallet.cpp:534]: A bit unusual, but hey, why not have a link to info :)

File: src/wallet/wallet2.cpp [hunk 3]
```diff
@@ -3613,7 +3587,7 @@ void wallet2::refresh(bool trusted_daemon, uint64_t start_height, uint64_t & blo
       THROW_WALLET_EXCEPTION_IF(!waiter.wait(), error::wallet_internal_error, "Exception in thread pool");
       throw;
     }
-    catch (const error::payment_required&)
+    catch (const error::deprecated_rpc_access&)
     {
       // no point in trying again, it'd just eat up credits
```
Comment (rbrunner7) [inline src/wallet/wallet2.cpp:3592]: This comment does not really apply anymore, does it, with no more "credits" ...

File: src/wallet/wallet2.cpp [hunk 4]
```diff
@@ -4078,13 +4049,13 @@ boost::optional<wallet2::keys_file_data> wallet2::get_keys_file_data(const epee:
     json.AddMember("original_view_secret_key", value, json.GetAllocator());
   }
   
-  value2.SetInt(m_persistent_rpc_client_id ? 1 : 0);
+  value2.SetInt(0);
```
Comment (rbrunner7) [inline src/wallet/wallet2.cpp:4052]: Maybe comment in some way that the 3 RPC payment related values do not exist anymore / are not active anymore and are only written to make sure we don't have any compatibility problems?

**Review by rbrunner7 — COMMENTED** [review]
Had a go at it. Pretty clear of course because mostly deleting stuff.

**Comment by jeffro256** [comment]
@rbrunner7 Thanks for reviewing! I updated the PR with some of your comments. What did you mean by "you let a lot of RPC payment related stuff stand in the wallet RPC server"? Would you please be able to point me to that?

**Comment by rbrunner7** [comment]
> Would you please be able to point me to that?

Well, I did a simple search for "rpc_pay" in the file `wallet_rpc_server.cpp` and got quite a number of hits, in methods that seem to implement RPC pay related commands, e.g. `COMMAND_RPC_GET_BULK_PAYMENTS` or `COMMAND_RPC_GET_PAYMENTS`.

**Comment by jeffro256** [comment]
IIUC, those RPC endpoints are unrelated to the mine-to-use system. AFAIK, those are RPC commands for retrieving general purpose payment information.

**Comment by rbrunner7** [comment]
(Slaps forehead) Of course, you are right, that's purely coincidence that the string "rpc_pay" results there.

But then I am still a bit confused: How can it be that the RPC server has (almost) no RPC Pay related code in it? Does that mean that it doesn't implement that? Or is it enough for it to work that merely `wallet2` cares about it?

**Comment by jeffro256** [comment]
Haha... there's certain words that don't seem like words to me anymore: transaction, payment, mine, access, daemon, etc. They all kinda blur together.

> Does that mean that it doesn't implement that?

Yes, I believe so. I really think the only wallet that supported mine-to-use was simplewallet, and not by default either. `wallet2` had most of the code for mine-to-use, but there was also a non-trivial amount of logic in simplewallet which managed the process.

**Review by rbrunner7 — APPROVED** [review]
Looks good.

Let's shed a last tear for all the good code that mooo implemented for this feature, and enjoy the simplified wallet code :)

**Review by tobtoht — APPROVED** [review]
Reviewed. Thank you for working on this.

**Comment by jeffro256** [comment]
Squashed and rebased against master

**Review by tobtoht — APPROVED** [review]

**Comment by jeffro256** [comment]
Rebased against master again to resolve conflicts with #8076

**Comment by rbrunner7** [comment]
I had a look at the rebase changes as far as they merge code in from my own #8076 (probably the largest single source of conflicts), and compiled the rebased code for a quick check whether pool handling still works.

I could not find any problems, this gets an "ok" from me.

**Review by selsta — APPROVED** [review]

**Comment by jeffro256** [comment]
Rebased against master again to fix merge conflicts with #8698

**Comment by jeffro256** [comment]
To verify rebase:
```
git show b7820e4 > a.txt
git show d2a591d > b.txt
diff a.txt b.txt
```

**Pull request merged**